pkg/api/v1alpha1/cluster.go

```diff
@@ -256,6 +257,36 @@ func (c *Cluster) IsReconcilePaused() bool {
 	return false
 }
 
+func ValidateClusterName(clusterName string) error {
```

Comment:
Just a thought, but any reason we moved the functions to here instead? I believe we can rename those functions and still have them accessible from the `validateClusterConfigName` function below right?

Reply:
Originally I thought to just call the functions from the validate package as you ask, but it introduced a circular dependency within the project. So I moved them into api so they can go with other api cluster config validations that were already present and then just expose same back to the ValidateClusterNameArg function.

Reply:
As seen in the image below, if you try to import validations package into api to reference the functions you get circular dependency error in the project. Please let me know you thoughts or suggestions.

![image](https://user-images.githubusercontent.com/2659771/145410657-71f5d0f1-545b-4ee2-a8f6-4c7e0645957d.png)

Reply:
The circular dependency is coming from the test files in validations and kubectl.go since the reference back to api.

Reply:
Hmm looks like our validations package is taking on too many responsibilities. Right way might be to introduce an inputvalidations package or something similar, but not required for this PR.